miden-lib/asm/sat/layout.masm

```diff
@@ -70,11 +70,12 @@ const.CONSUMED_NOTE_NUM_PTR=1000
 
 # The offsets at which data of a consumed note is stored relative to the start of its data segment
 const.CONSUMED_NOTE_HASH_OFFSET=0
+const.CONSUMED_NOTE_CORE_DATA_OFFSET=1
 const.CONSUMED_NOTE_SERIAL_NUM_OFFSET=1
 const.CONSUMED_NOTE_SCRIPT_ROOT_OFFSET=2
 const.CONSUMED_NOTE_INPUTS_HASH_OFFSET=3
 const.CONSUMED_NOTE_VAULT_ROOT_OFFSET=4
-const.CONSUMED_NOTE_NUM_ASSETS_OFFSET=5
+const.CONSUMED_NOTE_METADATA_OFFSET=5
 const.CONSUMED_NOTE_ASSETS_OFFSET=6
```

Comment:
I wonder if we should arrange this a bit differently so that metadata and hash are next to each other. Specifically, it could go like this:
```
const.CONSUMED_NOTE_METADATA_OFFSET=0
const.CONSUMED_NOTE_HASH_OFFSET=1
const.CONSUMED_NOTE_CORE_DATA_OFFSET=2
const.CONSUMED_NOTE_SERIAL_NUM_OFFSET=2
const.CONSUMED_NOTE_SCRIPT_ROOT_OFFSET=3
const.CONSUMED_NOTE_INPUTS_HASH_OFFSET=4
const.CONSUMED_NOTE_VAULT_ROOT_OFFSET=5
const.CONSUMED_NOTE_ASSETS_OFFSET=6
```
It seems a bit cleaner and may be useful in the future for computing parent node of note metadata and hash.

But either way, I wouldn't do it in this PR. Maybe create an issue to come back to this later?

Reply:
https://github.com/0xPolygonMiden/miden-base/issues/76